packages/next/src/withTRPC.tsx

```diff
@@ -205,11 +204,13 @@ export function withTRPC<
           trpc: trpcProp,
         };
 
+        const reactDomServer = await import('react-dom/server');
+
         // Run the prepass step on AppTree. This will run all trpc queries on the server.
         // multiple prepass ensures that we can do batching on the server
         while (true) {
           // render full tree
-          await ssrPrepass(createElement(AppTree, prepassProps as any));
+          reactDomServer.renderToString(createElement(AppTree, prepassProps));
```

Comment:
@luixo can you have a try to patch your project with this here instead and see if everything works fine?

Reply:
if it does, please open a new PR to `main` with this, it's a bit simpler

Reply:
See [this comment](https://github.com/trpc/trpc/pull/3343#discussion_r1435306147) for explanation of a case it might fail.
Let me know if you still think we should migrate to `renderToString` on main.